### PR TITLE
Options for MPI Executable in Ini-File

### PR DESCRIPTION
**Pull request by SEhrm**
Added entries
* `mpi_executable`, e.g. `mpirun`,
* `mpi_flag_for_processes`, e.g. `-n`, and 
* `mpi_command_line_inputs`, e.g. `--bind-to core`

to pytest's ini-file.

### CODE REVIEW

**Review by IHuismann — COMMENTED** [review]
Hi @SEhrm ,

looks splendid :-)

One small suggestion w.r.t. naming, but I'd be ok with this going in as is.

Cheers,

IHuismann

File: src/pytest_isolate_mpi/_plugin.py [hunk 1]
```diff
@@ -39,53 +39,47 @@ class MPIConfiguration:
...
+    mpirun_command_line_inputs: list[str]
+
+    def __init__(
+            self,
+            mpi_executable: str = None,
+            flag_for_processes="-n",
+            mpi_command_line_inputs: list[str] = None
+    ):
+        if not mpi_executable:
+            self.mpirun_executable = self.__get_mpirun_executable()
```
[the remark this answers is not in the document]
Comment (SEhrm) [reply src/pytest_isolate_mpi/_plugin.py]: You're right, ``__deduce_mpirun_executable`` makes much more sense.

File: docs/usage.rst [hunk 2]
```diff
@@ -105,6 +105,43 @@ mpi_tmp_path
...
+    Additional command line inputs to run the MPI executable with.
+    By default, none are given.
+
+For example, the ``pytest.ini`` for execution on HPC clusters with Slurm
+could be::
+
+
+    # pytest.ini
+    mpi_executable = srun
+    mpi_command_line_inputs = -N 2 --account MySlrumAccount
```
[the remark this answers is not in the document]
Comment (SEhrm) [reply docs/usage.rst:135]: If for example a cluster has nodes capable of running 128 processes each, then
```
mpi_flag_for_processes = -n  # default
mpi_command_line_inputs = -N 2
```
ensures that tests with `ranks=256` can be launched. I agree that the documentation should include non-default examples and propose
```
mpi_flag_for_processes = -N
mpi_command_line_inputs = --ntasks-per-node 128
```
as example, that can be used if one wants to check if a test runs on one or more nodes using `ranks=[1, 2]`.

[the remark this answers is not in the document]
Comment (sebasgo) [reply docs/usage.rst:135]: Yup, that is better.

File: docs/usage.rst [hunk 3]
```diff
@@ -105,6 +105,43 @@ mpi_tmp_path
...
+
+``pytest-isolate-mpi`` can be configured through the ``pytest``
+`configuration file`_:
+
+mpi_executable
+    The mpi executable to launch the forked MPI environment with. If
+    none is given, ``pytest-isolate-mpi`` tries ``mpirun`` and
+    ``mpiexec``.
+
+mpi_flag_for_processes
```
[the remark this answers is not in the document]
Comment (SEhrm) [reply docs/usage.rst]: Refactored to `mpi_option_for_processes`

File: docs/usage.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (SEhrm) [reply docs/usage.rst]: Done

File: docs/usage.rst [hunk 4]
```diff
@@ -105,6 +105,43 @@ mpi_tmp_path
...
+    The command line flag of the MPI executable indicating the number of
+    processes, such that ``pytest-isolate-mpi`` can launch the MPI
+    environement with the appropriate number of processes as defined in
+    the ``mpi`` mark. Defaults to ``-n``.
+
+mpi_command_line_inputs
```
[the remark this answers is not in the document]
Comment (SEhrm) [reply docs/usage.rst]: Refactored to `mpi_command_line_args`.

File: src/pytest_isolate_mpi/_plugin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SEhrm) [reply src/pytest_isolate_mpi/_plugin.py]: Refactored to `__deduce_mpirun_executable`.

**Review by sebasgo — APPROVED** [review]

**Pull request merged**